hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/steps/StakePeriodChanges.java

```diff
@@ -136,6 +144,27 @@ public void process(
                 logger.error("CATASTROPHIC failure updating end-of-day stakes", e);
                 stack.rollbackFullStack();
             }
+            final var tssStore = newWritableTssStore(stack, config);
+            final var rosterStore = newWritableRosterStore(stack, config);
+            final var activeRoster = rosterStore.getActiveRoster();
+            final var candidateRoster = rosterStore.getCandidateRoster();
+            final List<RosterEntry> rosterEntries = new ArrayList<>();
+            Optional.ofNullable(activeRoster).ifPresent(ar -> rosterEntries.addAll(ar.rosterEntries()));
+            Optional.ofNullable(candidateRoster).ifPresent(cr -> rosterEntries.addAll(cr.rosterEntries()));
+            for (final var rosterEntry : rosterEntries) {
+                final var nodeId = rosterEntry.nodeId();
+                TssEncryptionKeys currentTssEncryptionKeys = tssStore.getTssEncryptionKeys(nodeId);
+                if (currentTssEncryptionKeys != null
```

Comment:
This `TssEncryptionKey` should be updated when we call `setCandidateRoster` and the `candidateRoster` has same node right ? From my understanding this logic should not be in this class

Reply:
I thought we always wanted to rotate the TSS Encryption Keys on the stake period boundary prior to starting to key the candidate roster? (Next tss public key becomes current in state)

Refactored this slightly.

Reply:
If we put it in `setCadidateRoster` it wouldn't allow for the keys to be rotated unless there is a new candidate roster. Depends what behavior we want.

Reply:
yeah , not sure either. I was in an impression that this should be done in `setCandidateRoster`. @tinker-michaelj  thoughts? 

Either way, we can fix this in next PR since it is behind feature flag.